### PR TITLE
Fix the version of protobuf and pin tools in the build image.

### LINKED ISSUE #1370: We don't know which gogoproto we are using
We don't pin it in the build container, so as far as I can tell we are using whichever one was in the Docker build cache for whoever built the build container.

This is a bit scary for debugging / reproducability.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #1370

### CODE REVIEW

File: build-image/Dockerfile [hunk 1]
```diff
@@ -8,16 +8,14 @@ RUN curl -fsSLo shfmt https://github.com/mvdan/sh/releases/download/v1.3.0/shfmt
...
 		golang.org/x/lint/golint \
-		github.com/kisielk/errcheck \
-		github.com/mjibson/esc \
-		github.com/client9/misspell/cmd/misspell \
-		github.com/jteeuwen/go-bindata/go-bindata \
-		github.com/golang/protobuf/protoc-gen-go \
-		github.com/gogo/protobuf/protoc-gen-gogoslick \
-		github.com/gogo/protobuf/gogoproto && \
+		github.com/kisielk/errcheck@v1.2.0 \
+		github.com/client9/misspell/cmd/misspell@v0.3.4 \
```
Comment (bboreham) [inline build-image/Dockerfile]: why did we drop `esc`?

File: build-image/Dockerfile [hunk 2]
```diff
@@ -8,16 +8,14 @@ RUN curl -fsSLo shfmt https://github.com/mvdan/sh/releases/download/v1.3.0/shfmt
...
+		github.com/golang/protobuf/protoc-gen-go@v1.3.1 \
+		github.com/gogo/protobuf/protoc-gen-gogoslick@v1.2.1 \
```
Comment (bboreham) [inline build-image/Dockerfile]: why did we drop `go-bindata` ?

File: pkg/chunk/storage/caching_index_client.pb.go [hunk 3]
```diff
@@ -420,13 +420,8 @@ func (this *ReadBatch) String() string {
...
 		return "nil"
 	}
-	repeatedStringForEntries := "[]Entry{"
-	for _, f := range this.Entries {
-		repeatedStringForEntries += strings.Replace(strings.Replace(f.String(), "Entry", "Entry", 1), `&`, ``, 1) + ","
-	}
-	repeatedStringForEntries += "}"
 	s := strings.Join([]string{`&ReadBatch{`,
-		`Entries:` + repeatedStringForEntries + `,`,
+		`Entries:` + strings.Replace(strings.Replace(fmt.Sprintf("%v", this.Entries), "Entry", "Entry", 1), `&`, ``, 1) + `,`,
```
Comment (bboreham) [inline pkg/chunk/storage/caching_index_client.pb.go]: This change is removing https://github.com/gogo/protobuf/pull/557 - do we want to go backwards?

File: build-image/Dockerfile (hunk 1 above)
Comment (tomwilkie) [reply build-image/Dockerfile]: We don't use it AFAICT.

File: build-image/Dockerfile (hunk 2 above)
Comment (tomwilkie) [reply build-image/Dockerfile]: We don't use it AFAICT.

File: pkg/chunk/storage/caching_index_client.pb.go (hunk 3 above)
Comment (tomwilkie) [reply pkg/chunk/storage/caching_index_client.pb.go]: There is no gogo release with that fix in it. 

We're not using the String function for any production use (I don't think displaying the proto on /ring is affected adversely.)

**Comment by tomwilkie** [comment]
@bboreham do those responses assuage your concerns?

**Pull request merged**